homeassistant/components/environment_canada/sensor.py

```diff
@@ -27,94 +208,67 @@ def validate_station(station):
-        self._attr_name = f"{coordinator.config_entry.title} {label}"
-        self._attr_unique_id = f"{self.ec_data.metadata['location']}-{sensor}"
-        self._attr = None
-        self._unit = None
-        self._device_class = None
 
-    @property
-    def extra_state_attributes(self):
-        """Return the state attributes of the device."""
-        return self._attr
-
-    @property
-    def native_unit_of_measurement(self):
-        """Return the units of measurement."""
-        return self._unit
-
-    @property
-    def device_class(self):
-        """Return the class of this device, from component DEVICE_CLASSES."""
-        return self._device_class
+class ECSensor(ECBaseSensor):
+    """Environment Canada sensor for conditions."""
 
     @property
     def native_value(self):
         """Update current conditions."""
-        metadata = self.ec_data.metadata
-        sensor_data = self.ec_data.conditions.get(self.sensor_type)
-        if not sensor_data:
-            sensor_data = self.ec_data.alerts.get(self.sensor_type)
-
-        self._attr = {}
-        value = sensor_data.get("value")
-
-        if isinstance(value, list):
-            state = " | ".join([str(s.get("title")) for s in value])[:255]
-            self._attr.update(
-                {ATTR_TIME: " | ".join([str(s.get("date")) for s in value])}
-            )
-        elif self.sensor_type == "tendency":
-            state = str(value).capitalize()
+        sensor_type = self.entity_description.key
+        if sensor_type == "timestamp":
+            return self._ec_data.metadata.get("timestamp")
+
+        value = self._ec_data.conditions.get(sensor_type, {}).get("value")
+        if sensor_type == "tendency":
+            value = str(value).capitalize()
         elif isinstance(value, str) and len(value) > 255:
-            state = value[:255]
+            value = value[:255]
             _LOGGER.info(
                 "Value for %s truncated to 255 characters", self._attr_unique_id
             )
-        else:
-            state = value
-
-        if sensor_data.get("unit") == "C" or self.sensor_type in (
-            "wind_chill",
-            "humidex",
-        ):
-            self._unit = TEMP_CELSIUS
-            self._device_class = SensorDeviceClass.TEMPERATURE
-        else:
-            self._unit = sensor_data.get("unit")
-
-        self._attr.update(
-            {
-                ATTR_OBSERVATION_TIME: metadata.get("timestamp"),
-                ATTR_LOCATION: metadata.get("location"),
-                ATTR_STATION: metadata.get("station"),
-            }
-        )
-        return state
+        return value
+
+
+class ECAlertSensor(ECBaseSensor):
+    """Environment Canada sensor for alerts."""
+
+    @property
+    def native_value(self):
+        """Return the state."""
+        alert_name = self.entity_description.key
+        value = self._ec_data.alerts.get(alert_name, {}).get("value")
+
+        self._attr_extra_state_attributes = {
```

Comment:
Hmm, this is a bit of a cross-over, maybe implement the good old `def extra_state_attributes` property method instead?

Reply:
Done